### PR TITLE
Main screen state

### PR DESCRIPTION
**Pull request by xor3r**
- Added local state (graphics settings)
- Added global state (online status)
- Fixed some syntax

### CODE REVIEW

**Comment by damoklov** [comment]
## Local state (game settings)

  


## Global state (online/offline)

File: lib/main_view/element/row_with_status.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+Widget statusShow() {
+  return StoreConnector<StatusState, String>(
+    converter: (store) => store.state.status.toString(),
+    builder: (context, viewModel) {
+      return Center(
+        heightFactor: 2,
+        child: Text(
+            viewModel == 'true' ? 'Online' : 'Offline',
```
Comment (igor-leshkevych) [inline lib/main_view/element/row_with_status.dart:30]: 1. don't use string `'true'`, use boolean
2. don't use name `viewModel`, use something meaningful (`isOnline`), etc.

File: lib/main_view/home.dart [hunk 2]
```diff
@@ -26,9 +29,93 @@ Widget _textContainer(
...
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _toggleEffects() {
+    setState(() {
+      if (_effects) {
```
Comment (igor-leshkevych) [inline lib/main_view/home.dart:77]: just use `_effects = !_effects`

File: lib/states.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import 'package:meta/meta.dart';
+
+@immutable
+class StatusState {
+  static var empty = StatusState(false);
+  bool status;
+  StatusState(this.status);
+}
```
Comment (igor-leshkevych) [inline lib/states.dart:8]: don't forget empty lines

File: pubspec.yaml [hunk 4]
```diff
@@ -29,7 +29,8 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-
+  flutter_redux: ^0.8.2
+  redux: ^5.0.0
```
Comment (igor-leshkevych) [inline pubspec.yaml:33]: It's a brave decision to use Redux

File: lib/states.dart [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import 'package:meta/meta.dart';
+
+@immutable
+class StatusState {
+  static var empty = StatusState(false);
```
Comment (igor-leshkevych) [inline lib/states.dart:5]: use final

File: lib/states.dart [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  bool status;
```
Comment (igor-leshkevych) [inline lib/states.dart:6]: marked as `@immutable` but is actually mutable, use const/final here, too

File: lib/reducers.dart [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import 'actions.dart';
+import 'states.dart';
+
+StatusState statusReducer(StatusState previousState, dynamic action) {
+  if (action is OnlineAction) {
```
Comment (igor-leshkevych) [inline lib/reducers.dart:5]: use switch instead:
```
switch (action.runtimeType) {
    case OnlineAction:
      return StatusState(!action.status);
    case OfflineAction:
      return StatusState(action.status);
  }

  return previousState;
```

**Pull request merged**